feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/domain/StakingInteractor.kt

```diff
@@ -219,16 +223,13 @@ class StakingInteractor(
                     stakingRepository.ledgerFlow(stash),
                     stakingRepository.observeActiveEraIndex(networkType)
                 ) { ledger, activeEraIndex ->
-                    val erasPerDay = networkType.runtimeConfiguration.erasPerDay
                     val unbondingDuration = stakingConstantsRepository.lockupPeriodInEras()
 
                     ledger.unlocking
                         .filter { it.isUnbondingIn(activeEraIndex) }
                         .map {
-                            val createdAtEra = it.era - unbondingDuration
-                            val relativeInfo = eraRelativeInfo(createdAtEra, activeEraIndex, unbondingDuration, erasPerDay)
-
-                            Unbonding(it.amount, relativeInfo.daysLeft)
+                            val leftTime = stakingRepository.eraLeftTime(destinationEra = it.era)
```

Comment:
Here, `stakingRepository.eraLeftTime` is requested `N` times, where N is the number of payouts. Meaning, there will be `4N` requests to the local storage (aka DB) in the RepositoryImpl. It seems to be unnecessary since the same data is fetched each time. 
My suggestion is to move calculation logic out of repository in the following way
```
interface StakingRepository {
// ..
    fun currentSessionIndex, currentSlot, genesisSlot, ...
}
class EraTimeCalculator(val currentSessionIndex: Bigint, .....) {
    fun calculateTime(era: EraIndex) {
        // calcluation here
    }
}

class EraTimeCalculatorFactory(val repository: StakingRepository) {
   fun create() :  EraTimeCalculator {
    // get slots, session e.t.c. from repository and create EraTimeCalculator
   }
}

This way db will be queried only once and also calculation logic will be separated from data retrieval

Reply:
Done